File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    prev = words[0]
    used = {prev}
    for i in range(1, len(words)):
        current = words[i]

        if prev[-1] != current[0] or current in used:
            return [i % n + 1, i // n + 1]

        used.add(current)
        prev = current
        
    return [0,0]

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <unordered_set>

using namespace std;

vector<int> solution(int n, vector<string> words) {
    string prev = words[0];
    unordered_set<string> used;

    used.insert(prev);

    for (int i = 1; i < words.size(); i++) {
        string current = words[i];

        if (prev.back() != current.front() || used.count(current)) {
            return {(i % n) + 1, (i / n) + 1};
        }

        used.insert(current);
        prev = current;
    }

    return {0, 0};
}